pkg/component/controller/apiendpointreconciler.go

```diff
@@ -111,15 +113,15 @@ func (a *APIEndpointReconciler) reconcileEndpoints(ctx context.Context) error {
 
 	externalAddress := a.externalAddress
 
-	ipAddrs, err := a.resolver.LookupIPAddr(ctx, externalAddress)
+	ipAddrs, err := a.resolver.LookupIP(ctx, a.afnet, externalAddress)
```

Comment:
In the light of this: Maybe we should use the address family type for the afnet field here. And maybe we should call the empty address family "Unspecified".

We could either add a switch here to get the right network value for the lookup call right here:

```go
network := "ip"
switch n.AddressFamily {
case AddressFamilyIPv4:
	network = "ipv4"
case AddressFamilyIPv6:
	network = "ipv6"
}
```

or add the above as a method to the AddressFamily type:

```go
ipAddrs, err := a.resolver.LookupIP(ctx, a.addressFamily.ipNetwork(), externalAddress)
```